tools/cdk-build-tools/bin/cdk-test.ts

```diff
@@ -94,3 +94,49 @@ main().then(() => {
   process.stderr.write(`!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n`);
   process.exit(1);
 });
+
+export type Obj<T> = { [key: string]: T };
+
+/**
+ * Deep merge two objects into one.
+ * Arrays are *not* deep merged, they're overwritten instead.
+ * An undefined key in o2 results in an undefined key in the result.
+ */
+function deepMerge(o1: Obj<any>, o2: Obj<any>): Obj<any> {
```

Comment:
I suppose there's no standard deepmerge in the node libraries.

Let's extract this into `lib/` and write a test for it.